### PR TITLE
boa: add dependence check before install

### PR DESCRIPTION
**Pull request by FeelyChau**
add dependence check before install.

### CODE REVIEW

File: packages/boa/tools/check-dependence.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+const { run, PLATFORM } = require('./utils');
+
+if (process.version < 'v12.0.0') {
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:4]: We can not compare strings.

Comment (FeelyChau) [reply packages/boa/tools/check-dependence.js]: Done.

File: packages/boa/tools/check-dependence.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const version = process.version.split('.')[0];
+
+if (parseInt(version.substr(1)) < 12) {
+  throw new TypeError(`require Node.js >= v12.0.0, but ${process.version} found`);
+}
+
+let cmds = [ 'rm', 'make', 'tar' ];
+
+if (PLATFORM === 'linux') {
+  cmds = cmds.concat([ 'wget' ]);
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:12]:
```suggestion
  cmds.push('wget');
```

And we can declare the `cmds` in constant.

File: packages/boa/tools/check-dependence.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+} else if (PLATFORM === 'darwin') {
+  cmds = cmds.concat([ 'curl' ]);
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:14]: Ditto

File: packages/boa/tools/check-dependence.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+} else {
+  throw new TypeError(`No support for your platform ${PLATFORM}`);
+}
+
+cmds.forEach(cmd => {
+  console.log('checking:');
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:20]: Please remove the logs and it's unnecessary IMO.

File: packages/boa/tools/check-dependence.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+const { run, PLATFORM } = require('./utils');
+const version = process.version.split('.')[0];
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:3]:
```suggestion
const majorVersion = process.version.split('.')[0];
```

**Comment by yorkie** [comment]
@FeelyChau Please fix the conflicting issue.

**Comment by FeelyChau** [comment]
> @FeelyChau Please fix the conflicting issue.

Done.

File: packages/boa/tools/check-dependence.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  throw new TypeError(`No support for your platform ${PLATFORM}`);
+}
+
+cmds.forEach(cmd => {
+  try {
+    run(`command -v ${cmd}`);
+  } catch (err) {
+    throw new TypeError(`command ${cmd} check failed: ${err}`);
+  }
+})
```
Comment (yorkie) [inline packages/boa/tools/check-dependence.js:25]:
```suggestion
});
```

**Pull request merged**